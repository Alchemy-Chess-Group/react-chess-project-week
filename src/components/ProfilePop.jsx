import style from './ProfilePop.css';
import React, { useState } from 'react';

export default function ProfilePop(profile) {
  const [pop, setPop] = useState(false);
  const { name, displayName, email, bio } = profile.profile;
  console.log(profile, name);

  const togglePop = () => {
    setPop(!pop);
  };
  return (
    <div>
      <p onClick={togglePop}>{displayName}</p>
      {pop && (
        <div className={style.popupBox}>
          <div className={style.box}>
            <span className={style.closeIcon} onClick={togglePop}>
              x
            </span>
            <div>{name}</div>
            <div>{displayName}</div>
            <div>{email}</div>
            <div>{bio}</div>
          </div>
        </div>
      )}
    </div>
  );
}
