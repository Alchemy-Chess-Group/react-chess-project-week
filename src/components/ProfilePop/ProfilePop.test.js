import { render } from '@testing-library/react';
import ProfilePop from './ProfilePop';

const profile = {
  profile: {
    name: 'James',
    displayName: 'Jimmothy',
    email: 'james.armen@gmail.com',
    bio: 'This is a bio',
  },
};

test('Tests that the profile popup renders', () => {
  const { container } = render(<ProfilePop profile={profile} />);

  expect(container).toMatchSnapshot();
});
