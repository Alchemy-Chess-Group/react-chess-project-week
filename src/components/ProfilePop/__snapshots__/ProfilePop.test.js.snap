// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests that the profile popup renders 1`] = `
<div>
  <div>
    <p />
  </div>
</div>
`;
